smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs

```diff
@@ -78,57 +75,70 @@ pub(crate) fn execute_any_deposit(
         );
     }
 
-    // Swap logic
-    // TODO_FUTURE: Optimize this if conditions
-    let (swap_amount, swap_direction, left_over_amount) = if !deposit_info.base_refund.is_zero() {
-        // range is above current tick
-        let swap_amount = if pool_details.current_tick > position.upper_tick {
-            deposit_info.base_refund
+    let twap_price = get_twap_price(deps.storage, &deps.querier, &env, 24u64)?;
+    let (offer, out_denom, remainder, price) = if !deposit_info.base_refund.is_zero() {
```

Comment:
If we change offer according to Stefano's comments, lets change it here to so we have token_in, out_denom

Reply:
done